### ISSUE TITLE
console.error: "Image source size {28.666666666666668, 25} does not match loaded image size {28, 25}."

### ISSUE THREAD

**Issue by jasonmerino** [facebook/react-native#5586]
I'm getting this error when loading up the main view of the app I'm working on. I've tracked it down to one of the icon images in the tab bar. The dimensions of the image it's trying to load up is 56px x 50px. The funny thing is that it seems to only be happening on the @2x version of the image. The @3x image doesn't show this issue.

I've tried updating the image and resaving it to the exact size of 56 x 50 to no avail. Not sure if this is an issue with React Native, but figured I'd put it up here since the issue just popped up a couple versions ago when I upgraded the version of React Native.

![simulator screen shot jan 27 2016 1 17 14 pm](https://cloud.githubusercontent.com/assets/1660279/12628888/2e7360c0-c4fa-11e5-9d3e-a5ee14b2c992.png)

**Comment by coderdave** [comment]
I got this error too but only after upgrading to xcode 7.2.1

**Comment by jasonmerino** [comment]
So I just got a message from the designer I work with that he has resolved this issue. May have been the way the image was saved initially. Who knows.

Going to leave this open, but as of now, this issue has been resolved on our side.

**Comment by umfana** [comment]
I have the same issue. Commenting out the section in RCTConvert.m is a workaround for me at the moment: https://github.com/facebook/react-native/blob/master/React/Base/RCTConvert.m#L914-L918 but obviously that is not a suitable fix and it may have side effects.

**Comment by futurefeeling** [comment]
anyone fixed that problem? I meet the same problem.

**Comment by caoyue** [comment]
I had a image that  57px × 57px  for @2x and 85px × 85px for @3x, met the same problem.
Guess that 57/2 != 85/3 cause this problem? so I updated the images to 56px × 56px and 84px × 84px, and it was resolved :)

**Comment by MrPluto** [comment]
why this error occured? i got this error in debug but not in release. it's very werid

**Comment by geirman** [comment]
@facebook-github-bot bugfix

**Comment by geirman** [comment]
@facebook-github-bot label Help Wanted

**Comment by joeytat** [comment]
I've got the same error message and found out this is because my 3x.png image are not exactly thrice the point size.

**Comment by donald-jackson** [comment]
Same error for me, only able to get it to work by having exactly 3* and 2* multiples of image sizes.

**Comment by objectivesheep** [comment]
Same issue. What solved it for me is having a 3x image that doesn't have an odd height or width (e.g. 80x70 is ok but 80x69 is an issue) and then generate the 2x and 1x from that. It appears to be an issue with decimals... apparently react native wants whole numbers :-)

**Comment by piv199** [comment]
The same using NavigatorIOS->initalRoute->titleImage. I've just got image from xcode resources and past it to react native project, made reference to the image. And it doesn't display it :( Why it's not automatically resized or why this error is ever shown?

**Comment by reactionic127** [comment]
I found this solution.
I've got this issue in Tab Bar icon image.
It was my problem. (home.png(25*25), home@2x.png(50*51), home@3x.png(75*76))
I solved this issue by resizing home@2x.png and home@3x.png file.
This is good result.
25*25, 50*50, 75*75
Hope this helps.
Regards.

**Comment by anncatton** [comment]
I've been getting this issue with the thumbImage for a Slider, but all my images are sized with matching ratios: 1x(40x40), 2x(80x80) and 3x(120x120). But my error says the loaded image size is {0,0} so I'm not sure if it's the same as the above issues?
![handleslidererror](https://cloud.githubusercontent.com/assets/5624601/26171342/9bfcdf4c-3b12-11e7-8e20-21c2305de2e1.png)

Anyone else experienced this? It's happened in RN 0.39 and RN 0.44, but occurs only intermittently.

**Comment by objectivesheep** [comment]
The sizes need to match and be perfect multiplications in their 2x and 3x
forms. If you have img at 20*20 make sure that your 2@ and 3@ are exact
multiplications. If not rn will complain. It might be any one image in the
triplet... not necessarily the 1x

Op wo 17 mei 2017 om 21:21 schreef Ann Catton <notifications@github.com>

**Comment by anncatton** [comment]
The error complains about the 2x image, although the size it gives is the 1x dimensions. All 3 are definitely exact multiples - 40, 80 and 120.

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/consoleerror-image-source-size-28666666666666668-25-does-not-match-loaded-image-

**Comment by melvynkim** [comment]
> The error complains about the 2x image, although the size it gives is the 1x dimensions. All 3 are definitely exact multiples - 40, 80 and 120.

I'm having exact same issue.

**Comment by WhoJave** [comment]
@anncatton Have you solved this error ? It occurs only intermittently,too ,My react-native version is 0.49.3

**Comment by getaaron** [comment]
@hramos I still see this issue in React Native 0.55. Can we reopen the issue?

**Comment by amit2908** [comment]
@anncatton how did you managed to solve the issue which is giving {0,0} size of loaded image.

**Comment by anncatton** [comment]
@amit2908 I never did find a solution for this issue, but I haven't had time to look at it recently.

**Comment by pribeh** [comment]
@getaaron I'm running into this a bunch right now. Any idea how to resolve this? Have you?

**Comment by hramos** [comment]
You can resolve this by making sure your 2x and 3x images are exact multiples of the 1x image size.

**Comment by pribeh** [comment]
@hramos thanks. I will make sure to do that.

**Comment by getaaron** [comment]
@hramos I don’t think that fixes the part of the issue where some images load as 0,0 sized images

**Comment by pribeh** [comment]
@getaaron that's the error message we're receiving. Perhaps we should create a new issue for this?

**Comment by rodrigoelp** [comment]
Following this, does anybody has steps to reproduce?

**Comment by mccordgh** [comment]
__If you need to scan the QR Code on iOS you don't need to do it through the Expo app:__
  1. Open the Camera app
  2. Point the camera at the QR Code
  3. It should recognize it and give you a banner notification to open in Expo
  4. Click the banner notification

**Comment by Niryo** [comment]
@anncatton , are you using custom root with the metro bundler?
I have the same problem only in a project with custom root.

**Comment by TheSavior** [comment]
The solution to the issue reported by the OP is that the 1x, 2x, and 3x images must be exact multiples of the 1x image. 

It seems like the 0,0 sized image error is a separate problem. I'm going to close this as it seems like the original issue is resolved, but feel free to create a new issue for the 0,0 problem.

**Comment by rupalpatel0008** [comment]
> why this error occured? i got this error in debug but not in release. it's very werid

@MrPluto We are also facing the same issue in debug mode.
